src/Exporter/DataItemToElementMapper.php

```diff
@@ -0,0 +1,171 @@
+<?php
+
+namespace SMW\Exporter;
+
+use SMW\Exporter\Element\ExpResource;
+use SMW\Exporter\Element\ExpNsResource;
+use SMW\Exporter\Element\ExpElement;
+use SMW\Exporter\Element\ExpLiteral;
+use SMWExporter as Exporter;
+use SMWDataItem as DataItem;
+use SMWDITime as DITime;
+use RuntimeException;
+
+/**
+* @license GNU GPL v2+
+* @since 2.2
+*
+* @author mwjames
+* @author Markus Krötzsch
+*/
+class DataItemToElementMapper {
+
+	/**
+	 * @var array
+	 */
+	private $dataItemToElementMappers = array();
+
+	/**
+	 * @since 2.2
+	 *
+	 * @param $id integer
+	 * @param Closure $dataItemToExpElementMapper
+	 */
+	public function registerDataItemToExpElementMapper( $id, \Closure $dataItemToExpElementMapper ) {
+		$this->dataItemToElementMappers[$id] = $dataItemToExpElementMapper;
+	}
+
+	/**
+	 * Create an Element that encodes the data for the given dataitem object.
+	 * This method is meant to be used when exporting a dataitem as a subject
+	 * or object.
+	 *
+	 * @param DataItem $dataItem
+	 *
+	 * @return Element|null
+	 */
+	public function mapDataItemToElement( DataItem $dataItem ) {
+
+		if ( $this->dataItemToElementMappers === array() ) {
+				$this->initDataItemToElementMapper();
+		}
+
+		foreach ( $this->dataItemToElementMappers as $id => $dataItemToElementMapper ) {
+				if ( $id === $dataItem->getDIType() ) {
+					return $dataItemToElementMapper( $dataItem );
+				}
+		}
+
+		return null;
+	}
+
+	private function initDataItemToElementMapper() {
+
+		// Not used for now
+		$lang = '';
+
+		$this->registerDataItemToExpElementMapper( DataItem::TYPE_NUMBER, function( $dataItem ) use ( $lang ) {
+			return new ExpLiteral(
+					strval( $dataItem->getNumber() ),
+					'http://www.w3.org/2001/XMLSchema#double',
+					$lang,
+					$dataItem
+			);
+		} );
+
+		$this->registerDataItemToExpElementMapper( DataItem::TYPE_BLOB, function( $dataItem ) use ( $lang ) {
+			return new ExpLiteral(
+					smwfHTMLtoUTF8( $dataItem->getString() ),
+					'http://www.w3.org/2001/XMLSchema#string',
+					$lang,
+					$dataItem
+			);
+		} );
+
+		$this->registerDataItemToExpElementMapper( DataItem::TYPE_BOOLEAN, function( $dataItem ) use ( $lang ) {
+
+			$xsdvalue = $dataItem->getBoolean() ? 'true' : 'false';
+
+			return new ExpLiteral(
+					$xsdvalue,
+					'http://www.w3.org/2001/XMLSchema#boolean',
+					$lang,
+					$dataItem
+			);
+		} );
+
+		$this->registerDataItemToExpElementMapper( DataItem::TYPE_URI, function( $dataItem ) {
+			return new ExpResource(
+					$dataItem->getURI(),
+					$dataItem
+			);
+		} );
+
+		$this->registerDataItemToExpElementMapper( DataItem::TYPE_TIME, function( $dataItem ) use ( $lang ) {
```

Comment:
For this one you are better off putting the implementation in a class, so you can split the function